src/oop/object_interactions/medium/q3/MainTest.java

```diff
@@ -0,0 +1,218 @@
+                "Your " + makeCopyMethodName + " method does not make a copy of the " + myMonthAttributeName + " attribute correctly.");
+        _assertEquals(day, testClass.getFieldValue(classInstance, myDayAttributeName),
+                "Your " + makeCopyMethodName + " method does not make a copy of the " + myDayAttributeName + " attribute correctly.");
+        _assertEquals(story, testClass.getFieldValue(classInstance, myStoryAttributeName),
+                "Your " + makeCopyMethodName + " method does not make a copy of the " + myStoryAttributeName + " attribute correctly.");
+    }
+
+    @ParameterizedTest
+    @MethodSource("inputProvider")
+    @Tag("dependency1")
+    public void newspaperClassHasCorrectToStringMethod(int year, int month, int day, String story) throws Throwable {
+        Object[][] arguments = {
+                {year, int.class},
+                {month, int.class},
+                {day, int.class},
+                {story, String.class}
+        };
+        Object classInstance = testClass.createInstance(arguments);
+        String incorrectMethodDefinition = "The %s method in the %s class is not defined correctly. Make sure it is declared, spelt correctly, and has the correct parameters.";
+        String incorrectModifierMessage = "The %s method in the %s class has the wrong visibility modifier.";
+        String incorrectReturnType = "The %s method in the %s class has the incorrect return type.";
+        assertTrue(testClass.hasMethod("toString", null), String.format(incorrectMethodDefinition, "toString", className));
+        assertTrue(testClass.hasModifier("toString", null, "public"), String.format(incorrectModifierMessage, "toString", className));
+        assertTrue(testClass.hasReturnType("toString", null, String.class), String.format(incorrectReturnType, "toString", className));
+        String[] methodModifiers = {"public"};
+        String expected = year + "/" + month + "/" + day + ":The story is: " + story;
+        String incorrectToStringMessage = String.join(" ",
+                "Your", className, "toString method does not return the correct String.");
+        Object output = testClass.callMethod("toString", methodModifiers, classInstance);
+        _assertEquals(expected, output, incorrectToStringMessage);
+    }
+
+    @Test
+    @Tag("dependency1")
+    @Tag("dependency2")
+    public void correctMainMethod() throws Throwable {
+        Clause[] clauses = {
+                new StringLiteral("2021/12/5:The story is: The votes are in!"),
+                new Optional(new StringLiteral(" ")),
+                new NewLine(),
+                new StringLiteral("-1/-1/-1:The story is:"),
+                new Optional(new StringLiteral(" ")),
+                new NewLine(),
+                new StringLiteral("2021/12/5:The story is: The votes are in!"),
+                new Optional(new StringLiteral(" "))
+        };
+        Object classInstance = classObject.createInstance();
```

Comment:
Please add a custom error message to specify the newspaper toString output was incorrect.